specs/consensus.md

```diff
@@ -71,16 +86,160 @@ Consensus Rules
+
+### `block.availableData.transactionData`
+
+Once [evidence has been processed](#blockavailabledataevidencedata), transactions are applied to the state. Note that _transactions_ mutate the state (essentially, the validator set and minimal balances), while _messages_ do not. See [the architecture documentation](./architecture.md) for more info.
+
+`block.availableData.transactionData` is simply a list of [WrappedTransaction](./data_structures.md#wrappedtransaction)s. For each wrapped transaction in this list, `wrappedTransaction`, with index `i` (starting from `0`), the following checks must be `true`:
+
+1. `wrappedTransaction.index` == `i`.
+
+For `wrappedTransaction`'s [transaction](./data_structures.md#transaction) `transaction`, the following checks must be `true`:
+
+1. `transaction.signature` must be a [valid signature](./data_structures.md#public-key-cryptography) over `transaction.signedTransactionData`.
+
+Finally, each `wrappedTransaction` is processed depending on [its transaction type](./data_structures.md#signedtransactiondata). These are specified in the next subsections, where `tx` is short for `transaction.signedTransactionData`, and `sender` is the recovered signing [address](./data_structures.md#address). After applying a transaction, the new state state root is computed.
+
+#### SignedTransactionDataTransfer
+
+The following checks must be `true`:
+
+1. `tx.type` == [`TransactionType.Transfer`](./data_structures.md#signedtransactiondata).
+1. `tx.amount` <= `state.accounts[sender].balance`.
+1. `tx.nonce` == `state.accounts[sender].nonce + 1`.
+
+Apply the following to the state:
 
-A transaction `tx` that requests a new validator initializes a new [Validator](data_structures.md#validator) leaf in the inactive validators subtree for that account as follows:
 ```
+state.accounts[sender].nonce += 1
+
+state.accounts[sender].balance -= tx.amount
+state.accounts[tx.to].balance += tx.amount
+```
+
+#### SignedTransactionDataPayForMessage
+
+The following checks must be `true`:
+
+1. `tx.type` == [`TransactionType.PayForMessage`](./data_structures.md#signedtransactiondata).
+1. `tx.nonce` == `state.accounts[sender].nonce + 1`.
+1. The `ceil(tx.messageSize / SHARE_SIZE)` shares starting at index `wrappedTransactions.messageStartIndex` must:
+    1. Have namespace ID `tx.messageNamespaceID`.
+1. `tx.messageShareCommitment` == computed as described [here](./data_structures.md#signedtransactiondatapayformessage).
+
+Apply the following to the state:
+
+```
+state.accounts[sender].nonce += 1
+```
+
+#### SignedTransactionDataPayForPadding
+
+1. `tx.type` == [`TransactionType.PayForPadding`](./data_structures.md#signedtransactiondata).
+1. The `ceil(tx.messageSize / SHARE_SIZE)` shares starting at index `wrappedTransactions.messageStartIndex` must:
+    1. Have namespace ID `tx.messageNamespaceID`.
+
+Apply the following to the state:
+
+```
+```
```

Comment:
Shouldn't we remove this type together with the paragraph under Caveats?
(https://github.com/lazyledger/lazyledger-specs/pull/89/files#r531078625)

Reply:
I don't want to be overzealous with the scope of this PR, so I filed #98 to follow up on how padding should be handled. For now I'd rather just leave the PayForPadding tx type as it was. https://github.com/lazyledger/lazyledger-specs/pull/89#discussion_r531129652